src/Orleans/Core/GrainAttributes.cs

```diff
@@ -86,6 +87,37 @@ public sealed class ImmutableAttribute : Attribute
         }
     }
 
+    namespace MultiCluster
+    {
+        /// <summary>
+        /// base class for multi cluster registration strategies.
+        /// </summary>
+        public abstract class RegistrationAttribute : Attribute
```

Comment:
I moved the ` [AttributeUsage(AttributeTargets.Class)]` to the base class, since we are expecting all derived attributes to target classes only. This seems to work as intended.